src/Relation/Binary/Construct/Closure/Reflexive/Properties.agda

```diff
@@ -0,0 +1,59 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some properties of reflexive closures
+------------------------------------------------------------------------
+
+{-# OPTIONS --safe --without-K #-}
+
+module Relation.Binary.Construct.Closure.Reflexive.Properties where
+
+open import Data.Product as Prod
+open import Function
+open import Relation.Binary
+open import Relation.Binary.Construct.Closure.Reflexive
+import Relation.Binary.PropositionalEquality as PropEq
+open PropEq using (_≡_; refl)
+open import Relation.Nullary
+
+module _ {a b ℓ} {A : Set a} {B : Set b} {_~_ : Rel A ℓ} where
+    =[]⇒ : ∀ {p q} {P : Rel A p} {Q : Rel B q} {f} → P =[ f ]⇒ Q → Refl P =[ f ]⇒ Refl Q
+    =[]⇒ x [ x∼y ] = [ x x∼y ]
+    =[]⇒ x refl = refl
+
+    respˡ : ∀ {p} {P : REL A B p} → P Respectsˡ _~_ → P Respectsˡ (Refl _~_)
+    respˡ p-respˡ-~ [ x∼y ] = p-respˡ-~ x∼y
+    respˡ _ refl = id
+
+    respʳ : ∀ {p} {P : REL B A p} → P Respectsʳ _~_ → P Respectsʳ (Refl _~_)
+    respʳ = respˡ
+
+module _ {a ℓ} {A : Set a} {_~_ : Rel A ℓ} where
+    dec : Decidable {_} {A} _≡_ → Decidable _~_ → Decidable (Refl _~_)
```

Comment:
Writing this as `Decidable {A = A} _≡_` is a little neater and clearer what's going on.

Reply:
Done